packages/utreexo/src/stump/accumulator.cairo

```diff
@@ -13,7 +13,37 @@ pub impl StumpUtreexoAccumulatorImpl of StumpUtreexoAccumulator {
     fn verify(
         self: @UtreexoStumpState, proof: @UtreexoBatchProof, del_hashes: Span<felt252>
     ) -> Result<(), ByteArray> {
-        // TODO
+        if (*proof.targets).is_empty() {
+            return Result::Ok(());
+        };
+
+        let computed_roots: Span<felt252> = proof.compute_roots(del_hashes, *self.num_leaves)?;
+
+        let mut number_matched_roots: u32 = 0;
+
+        // Should we reverse *self.roots like in rustreexo to reduce the number of iteration?
+        for i in 0
+            ..computed_roots
+                .len() {
+                    for root in *self
```

Comment:
This is not the same as what rustreexo does: you need to check that the roots at the same height are equal (or both are empty), not that there are two identical roots in both sets